src/features/marketplace/process/commands/change-price-item/store-change-price-item-command.ts

```diff
@@ -0,0 +1,33 @@
+import ItemService from '../../../../leda-nft/services/item.service';
+import MarketplaceError from '../../enums/marketplace-error.enum';
+import ICommand from '../../interfaces/command.interface';
+import MarketplaceState from '../../types/marketplace-state';
+
+export default class StoreChangePriceItemCommand implements ICommand<MarketplaceState> {
+  private readonly itemService: ItemService;
+
+  constructor(_itemService: ItemService) {
+    this.itemService = _itemService;
+  }
+
+  async execute(state: MarketplaceState): Promise<MarketplaceState> {
+    if (!state.itemId) return { ...state, error: MarketplaceError.RequiredItemId };
+    if (!state.price) return { ...state, error: MarketplaceError.RequiredPrice };
+    if (!state.marketplaceEvent) return { ...state, error: MarketplaceError.RequiredListEvent };
```

Comment:
Why do you need the marketplaceEvent here? And if you happen to need it, you are returning the wrong error.